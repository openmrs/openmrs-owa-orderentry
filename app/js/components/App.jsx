import React from 'react';

export default class App extends React.Component {
  constructor(props){
    super(props);

  }
  render() {
    return (
      <div>
        <h1>Hello, world!
        </h1>
      </div>
    );
  }
}
