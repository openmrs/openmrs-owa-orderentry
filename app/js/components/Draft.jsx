import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import shortid from 'shortid';
import constants from '../utils/constants';
import IconButton from './button/IconButton';

export class Draft extends PureComponent {
  renderDraftList = () => {
    let draftType;
    const { draftOrders, handleDraftDiscard } = this.props;
    return draftOrders.map((order) => {
      const isPanel = !!order.set;
      const isOtherOrderType = !!order.type;

      if (isPanel) {
        draftType = 'panel';
      } else {
        draftType = 'single';
      }
      if (isOtherOrderType) {
        draftType = order.type;
      }

      const orderName = order.display || order.drugName;
      const iconClass = classNames(
        'scale',
        {
          'i-gray': order.urgency === constants.ROUTINE || typeof order.urgency === 'undefined',
          'i-red': order.urgency === constants.STAT,
        },
      );

      return (
        <li className="draft-list small-font" key={shortid.generate()}>
          <span className="order-status">{!order.action ? 'NEW' : order.action}</span>
          <span className="draft-name">{orderName.toLowerCase()}</span>
          <div className="action-btn-wrapper">
            <span className="action-btn">
              { order.type !== 'drugorder' ?
                <IconButton
                  iconClass={iconClass}
                  iconTitle="Urgency"
                  dataContext={order}
                  onClick={this.props.toggleDraftLabOrderUrgency}
                  icon="&#x25B2;"
                  id="draft-toggle-btn icon-btn-anchor"
                /> :
                <IconButton
                  iconClass="icon-pencil"
                  iconTitle="EDIT"
                  dataContext={order}
                  onClick={this.props.editDraftDrugOrder}
                  id="draft-toggle-btn icon-btn-anchor"
                />
              }
            </span>
            <span className="action-btn right">
              <IconButton
                iconClass="icon-remove"
                iconTitle="Discard"
                id="draft-discard-btn"
                dataContext={{ order, draftType }}
                onClick={handleDraftDiscard}
              />
            </span>
          </div>
        </li>);
    });
  }

  render() {
    const { draftOrders, handleDraftDiscard, handleSubmit } = this.props;
    const numberOfDraftOrders = draftOrders.length;
    const isDisabled = !numberOfDraftOrders;
    return (
      <div className="draft-spacing draft-lab-layout">
        <h5 className="h5-draft-header">
          Unsaved Draft Orders ({numberOfDraftOrders})
        </h5>
        <div className="table-container">
          <ul className="draft-list-container">
            {this.renderDraftList()}
          </ul>
        </div>
        <br />
        <input
          type="button"
          id="draft-discard-all"
          onClick={() => handleDraftDiscard()}
          className="button cancel modified-btn"
          value={numberOfDraftOrders > 1 ? "Discard All" : "Discard"}
          disabled={isDisabled}
        />
        <input
          type="submit"
          onClick={handleSubmit}
          className="button confirm right modified-btn"
          value="Sign and Save"
          disabled={isDisabled}
        />
      </div>
    );
  }
}

Draft.propTypes = {
  draftOrders: PropTypes.arrayOf(PropTypes.any).isRequired,
  editDraftDrugOrder: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  handleDraftDiscard: PropTypes.func.isRequired,
  toggleDraftLabOrderUrgency: PropTypes.func.isRequired,
};

export default Draft;

