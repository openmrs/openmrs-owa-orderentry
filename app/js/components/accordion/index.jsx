import React from 'react';
import PropTypes from 'prop-types';

class Accordion extends React.Component {
  state = {
    isVisible: this.props.open === undefined ? false : this.props.open,
  };

  render() {
    return (
      <div className={`accordion ${this.props.border ? 'border' : ''}`}>
        <div className="header">
          <a
            role="button"
            tabIndex={0}
            onClick={() => {
              this.setState(() => ({ isVisible: !this.state.isVisible }));
            }}>
            <span>
              {
                this.state.isVisible ?
                  <i className="toggle-icon icon-caret-up small rotate180" /> :
                  <i className="toggle-icon icon-caret-up small rotate90" />
              }
            </span>
            {this.props.title}
          </a>
        </div>
        <div className={`content ${!this.state.isVisible ? 'close' : 'open'}`}>
          {this.props.children}
        </div>
      </div>
    );
  }
}


Accordion.propTypes = {
  border: PropTypes.bool,
  children: PropTypes.node.isRequired,
  open: PropTypes.bool,
  title: PropTypes.string.isRequired,
};

Accordion.defaultProps = {
  open: false,
  border: false,
};

export default Accordion;
