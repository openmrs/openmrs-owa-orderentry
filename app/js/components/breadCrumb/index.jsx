import React from 'react';
import PropTypes from 'prop-types';

const BreadCrumb = props => (
  <ul id="breadcrumbs">
    <li>
      <a href="../../">
        <i className="icon-home small" />
      </a>
    </li>
    <li>
      <i className="icon-chevron-right link" />
      <a
        href={`../../coreapps/clinicianfacing/patient.page?patientId=${props.patientId}`}
      >
        {props.name}
      </a>
    </li>
    <li>
      <i className="icon-chevron-right link" />
      Drug Orders
    </li>
  </ul>
);

BreadCrumb.propTypes = {
  name: PropTypes.string.isRequired,
  patientId: PropTypes.number,
};

BreadCrumb.defaultProps = {
  patientId: null,
};

export default BreadCrumb;
