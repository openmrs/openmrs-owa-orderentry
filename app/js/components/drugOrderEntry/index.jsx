import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Accordion } from '@openmrs/react-components';
import AddForm from './addForm/AddForm';
import PastOrders from './PastOrders';
import Tabs from '../tabs/Tabs';
import Tab from '../tabs/Tab';
import SearchDrug from '../searchDrug';
import ActiveOrders from './ActiveOrders';
import { setOrderAction } from '../../actions/orderAction';
import { deleteDraftOrder, deleteAllDraftOrders } from '../../actions/draftTableAction';
import DraftDataTable from './addForm/DraftDataTable';
import { selectDrugSuccess } from '../../actions/drug';
import './styles.scss';

export class SearchAndAddOrder extends React.PureComponent {
  state = {
    value: "",
    focused: false,
    editDrugUuid: '',
    editDrugName: '',
    editOrder: {},
    draftOrder: {},
  };

  componentDidMount() {
    const { selectedOrder, activity } = this.props;
    if (activity === 'EDIT' && !this.state.formattedDetails) {
      const details = (
        <td>
          {selectedOrder.drug.display}:
          {selectedOrder.dosingInstructions && ` ${selectedOrder.dosingInstructions}`}
          {(selectedOrder.quantity && selectedOrder.quantityUnits) && `, (Dispense: ${selectedOrder.quantity} ${selectedOrder.quantityUnits.display})`}
        </td>
      );
      this.handleEditActiveDrugOrder({ ...selectedOrder, status: activity }, details);
    }
  }

  onSelectDrug = (drugName) => {
    this.setState(() => ({
      value: drugName,
      focused: false,
    }));
  }

  onChange = (value) => {
    this.setState(() => ({
      value,
      focused: true,
    }));
  }

  handleDiscardOneOrder = (order) => {
    this.props.deleteDraftOrder(order);
    if (order.action === 'REVISE') {
      this.props.setOrderAction('DISCARD_EDIT', order.orderNumber);
    } else {
      this.props.setOrderAction('DISCARD_ONE', order.orderNumber);
    }
  }

  handleDiscardAllOrders = () => {
    this.props.deleteAllDraftOrders();
    this.props.setOrderAction('DISCARD_ALL', '0');
  }

  clearSearchField = () => {
    this.setState({
      value: "",
      focused: false,
      editDrugUuid: "",
      editDrugName: "",
    });
  }

  handleEditDraftOrder = (order) => {
    this.props.selectDrugSuccess(order.drug);
    this.setState({
      draftOrder: order,
      editDrugUuid: order.drug,
      editDrugName: order.drugName,
      orderNumber: order.orderNumber,
    }, () => {
      this.props.setOrderAction('EDIT', order.orderNumber);
    });
    this.props.deleteDraftOrder(order);
  }

  handleEditActiveDrugOrder = (order, details) => {
    let formattedDetails = details.props.children.join("");
    if (formattedDetails.length > 250) {
      formattedDetails = `${formattedDetails.substring(0, 250)}...`;
    }
    const editOrder = order;
    editOrder.action = "REVISE";
    this.setState({
      formattedDetails,
      editDrugUuid: order.drug.uuid,
      editDrugName: order.drug.display,
      editOrder,
      orderNumber: order.orderNumber,
    }, () => {
      this.props.setOrderAction('EDIT', order.orderNumber);
    });
  }

  removeOrder = () => {
    this.setState({
      editOrder: {},
      draftOrder: {},
    });
  }

  closeFormsOnTabChange = () => {
    this.clearSearchField();
    this.props.selectDrugSuccess();
    this.props.deleteAllDraftOrders();
  }

  renderSearchDrug = () => (
    this.state.editDrugName ?
      <p className="revise-order-padding">
        <b className="revised-order-text"> Revise for: {this.state.editDrugName} </b><br />
        <b className="current-order-color">Current Order: <em>{this.state.formattedDetails}</em></b>
      </p>
      :
      <SearchDrug
        value={this.state.value}
        focused={this.state.focused}
        onChange={this.onChange}
        onSelectDrug={this.onSelectDrug}
      />
  )

  renderAddForm = careSetting => (
    <div>
      <AddForm
        drugName={this.state.editDrugName ? this.state.editDrugName : this.props.drug.display}
        drugUuid={this.state.editDrugUuid ? this.state.editDrugUuid : this.props.drug.uuid}
        editOrder={this.state.editOrder}
        careSetting={careSetting}
        clearSearchField={this.clearSearchField}
        removeOrder={this.removeOrder}
        draftOrder={this.state.draftOrder}
        orderNumber={this.state.orderNumber}
      />
    </div>
  );

  renderDraftDataTable = careSetting => (
    (this.props.draftOrders.length > 0) &&
    <DraftDataTable
      draftOrders={this.props.draftOrders}
      handleDiscardOneOrder={this.handleDiscardOneOrder}
      handleDiscardAllOrders={this.handleDiscardAllOrders}
      handleEditDraftOrder={this.handleEditDraftOrder}
      careSetting={careSetting}
    />
  )

  render() {
    const {
      outpatientCareSetting, location, dateFormat,
    } = this.props;
    return (
      <div className="drug-order-entry">
        <h5 className="drug-form-header">New Drug Order</h5>
        {this.renderSearchDrug()}
        {this.renderAddForm(outpatientCareSetting)}
        {this.renderDraftDataTable(outpatientCareSetting)}
        <Accordion open title="Active Drug Orders">
          <ActiveOrders
            careSetting={outpatientCareSetting}
            location={location}
            dateFormat={dateFormat}
            handleEditActiveDrugOrder={this.handleEditActiveDrugOrder}
          />
        </Accordion>

        <Accordion title="Past Drug Orders">
          <PastOrders
            careSetting={outpatientCareSetting}
            dateFormat={dateFormat}
            location={location}
          />
          <br />
        </Accordion>
      </div>
    );
  }
}

const mapStateToProps = ({
  careSettingReducer:
  { outpatientCareSetting },
  drugSearchReducer,
  draftReducer: { draftDrugOrders },
  dateFormatReducer: { dateFormat },
  orderSelectionReducer: { activity, selectedOrder },
}) => ({
  outpatientCareSetting,
  drug: drugSearchReducer.selected,
  draftOrders: draftDrugOrders,
  dateFormat,
  activity,
  selectedOrder,
});

SearchAndAddOrder.defaultProps = {
  draftOrders: [],
  drug: null,
  selectedOrder: {},
  activity: '',
};

SearchAndAddOrder.propTypes = {
  drug: PropTypes.oneOfType([
    PropTypes.shape({
      uuid: PropTypes.string,
      display: PropTypes.string,
    }),
    PropTypes.string,
  ]),
  draftOrders: PropTypes.arrayOf(PropTypes.any),
  dateFormat: PropTypes.string.isRequired,
  location: PropTypes.shape({
    search: PropTypes.string,
  }).isRequired,
  setOrderAction: PropTypes.func.isRequired,
  deleteDraftOrder: PropTypes.func.isRequired,
  deleteAllDraftOrders: PropTypes.func.isRequired,
  outpatientCareSetting: PropTypes.shape({
    uuid: PropTypes.string.isRequired,
    display: PropTypes.string.isRequired,
  }).isRequired,
  selectDrugSuccess: PropTypes.func.isRequired,
  selectedOrder: PropTypes.object,
  activity: PropTypes.string,
};

export default connect(
  mapStateToProps,
  {
    setOrderAction,
    deleteDraftOrder,
    deleteAllDraftOrders,
    selectDrugSuccess,
  },
)(SearchAndAddOrder);
