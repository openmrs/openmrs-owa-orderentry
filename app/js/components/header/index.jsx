import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import View from './View';

import { fetchCurrentSession, fetchLocations, setCurrentLocation } from '../../actions/header';

export class Header extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      userDropdown: false,
      locationDropdown: false,
    };
  }

  componentWillMount() {
    this.props.fetchCurrentSession();
    this.props.fetchLocations();
  }

  toggleState = (key, value) => {
    this.setState(() => ({
      [key]: value || !this.state[key],
    }));
  }

  render() {
    return (
      <div>
        <View
          currentUser={this.props.currentUser}
          locations={this.props.locations}
          currentLocation={this.props.currentLocation}
          setCurrentLocation={this.props.setCurrentLocation}
          toggleState={this.toggleState}
          userDropdown={this.state.userDropdown}
          locationDropdown={this.state.locationDropdown}
        />
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  const { currentLocation, currentUser } = state.sessionReducer;
  const { locationTags } = state.locationReducer;
  return {
    currentLocation,
    currentUser,
    locations: locationTags,
  };
};

const actionCreators = {
  fetchCurrentSession,
  fetchLocations,
  setCurrentLocation,
};

Header.propTypes = {
  currentUser: PropTypes.string,
  currentLocation: PropTypes.shape().isRequired,
  fetchCurrentSession: PropTypes.func.isRequired,
  fetchLocations: PropTypes.func.isRequired,
  locations: PropTypes.array.isRequired,
  setCurrentLocation: PropTypes.func.isRequired,
};

Header.defaultProps = {
  currentUser: '',
};

export default connect(mapStateToProps, actionCreators)(Header);
