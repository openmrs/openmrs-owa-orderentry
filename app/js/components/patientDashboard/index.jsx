import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import View from './View';
import BreadCrumb from '../breadCrumb';
import { fetchPatientRecord, fetchPatientNote } from '../../actions/patient';

export class Dashboard extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showContactInfo: false,
    };
  }

  componentDidMount() {
    const query = new URLSearchParams(this.props.location.search);
    const patientUuid = query.get('patient');
    this.props.fetchPatientRecord(patientUuid);
    this.props.fetchPatientNote(patientUuid);
  }

  toggleDetailsView = () => {
    this.setState(() => ({
      showContactInfo: !this.state.showContactInfo,
    }));
  }

  render() {
    return (
      <div>
        <BreadCrumb
          name={`${this.props.patient.person.personName.familyName}. ${this.props.patient.person.personName.givenName}`}
          patientId={this.props.patient.patientId}
        />
        <View
          patient={this.props.patient}
          toggleDetailsView={this.toggleDetailsView}
          showContactInfo={this.state.showContactInfo}
          note={this.props.note}
        />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  patient: state.patientReducer.patient,
  note: state.noteReducer.note,
});

const actionCreators = {
  fetchPatientRecord,
  fetchPatientNote,
};

Dashboard.propTypes = {
  fetchPatientNote: PropTypes.func.isRequired,
  fetchPatientRecord: PropTypes.func.isRequired,
  location: PropTypes.shape().isRequired,
  note: PropTypes.array,
  patient: PropTypes.shape().isRequired,
};

Dashboard.defaultProps = {
  note: [],
};

export default connect(mapStateToProps, actionCreators)(Dashboard);
