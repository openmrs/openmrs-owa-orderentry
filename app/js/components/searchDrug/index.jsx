import React from 'react';
import { debounce } from 'lodash';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { searchDrug, selectDrug } from '../../actions/drug';
import View from './View';

export class SearchDrug extends React.Component {
  onOptionSelected = (drugUuid, drugName) => {
    this.props.selectDrug(drugUuid);
    this.props.onSelectDrug(drugName);
  }

  onChange = (event) => {
    event.preventDefault();
    const { target: { value } } = event;
    this.searchDrug(value);
    this.props.onChange(value);
  }

  search = debounce((text) => {
    this.props.searchDrug(text);
  }, 500);

  searchDrug = (text) => {
    if (text && text.trim().length > 2) {
      this.search(text);
    }
  }

  render() {
    return (
      <View
        searchDrug={this.searchDrug}
        focused={this.props.focused}
        onChange={this.onChange}
        onOptionSelected={this.onOptionSelected}
        results={this.props.drugSearch.drugs}
        loading={this.props.drugSearch.loading}
        error={this.props.drugSearch.error}
        value={this.props.value}
      />
    );
  }
}

const mapStateToProps = state => ({
  drugSearch: state.drugSearchReducer,
});

const actionCreators = {
  searchDrug,
  selectDrug,
};

SearchDrug.propTypes = {
  drugSearch: PropTypes.object.isRequired,
  searchDrug: PropTypes.func.isRequired,
  selectDrug: PropTypes.func.isRequired,
  onSelectDrug: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  focused: PropTypes.bool.isRequired,
  value: PropTypes.string,
};

SearchDrug.defaultProps = {
  value: '',
};

export default connect(mapStateToProps, actionCreators)(SearchDrug);
