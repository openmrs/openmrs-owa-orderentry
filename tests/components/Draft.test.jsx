import React from 'react';
import { Draft } from '../../app/js/components/Draft';

let props;
let mountedComponent;

props = {
  draftOrders: [
    { uuid: 6, display: 'hermatocrite', set: true, type: 'laborder' }
  ],
  handleSubmit: jest.fn(),
  handleDraftDiscard: jest.fn(),
};

const getComponent = () => {
  if (!mountedComponent) {
    mountedComponent = shallow(<Draft {...props} />);
  }
  return mountedComponent;
};

describe('Component: Draft', () => {
  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('should render on initial setup', () => {
    const component = getComponent();
    expect(component).toMatchSnapshot();
  });

  it('should display the action buttons', () => {
    const component = getComponent();
    const actionButton = component.find('.action-btn');
    expect(actionButton.length).toEqual(2);
  });

  it('Should be able to display drug data to the list', () => {
    const component = getComponent();
    component.setProps({ draftOrders: [
      { uuid: 6, drugName: 'paracetamol' }
    ]});
    const draftName = component.find('.draft-name');
    expect(draftName.exists()).toBeTruthy();
    expect(draftName.props().children).toEqual('paracetamol');
  });

  it('Should simulate the discard event', () => {
    const component = getComponent();
    component.setProps({ draftOrders: [
      { uuid: 6, drugName: 'paracetamol' },
      { uuid: 7, display: 'hermatology', action: 'NEW', type: 'drugorder'}
    ]});
    const discardButton = component.find('#draft-discard-all');
    discardButton.simulate('click', {});
    expect(props.handleDraftDiscard).toBeCalled();
  });
});

