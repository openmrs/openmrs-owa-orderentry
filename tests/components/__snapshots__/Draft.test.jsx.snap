// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Draft should render on initial setup 1`] = `
<Draft
  dispatch={
    [MockFunction] {
      "calls": Array [
        Array [
          [Function],
        ],
      ],
    }
  }
  draftOrders={
    Array [
      Object {
        "display": "hermatocrite",
        "names": Array [
          Object {
            "conceptNameType": "SHORT",
            "id": 13207,
            "locale": "en",
            "localePreferred": false,
            "name": "Hct",
            "uuid": "0b7ba816-15f5-102d-96e4-000c29c2a5d7",
            "voided": false,
          },
          Object {
            "conceptNameType": "FULLY_SPECIFIED",
            "id": 13204,
            "locale": "en",
            "localePreferred": true,
            "name": "Hematocrit",
            "uuid": "3e148d20-26fe-102b-80cb-0017a47871b2",
            "voided": false,
          },
        ],
        "set": true,
        "type": "laborder",
        "uuid": 6,
      },
    ]
  }
  editDraftDrugOrder={[MockFunction]}
  handleDraftDiscard={[MockFunction]}
  handleSubmit={[MockFunction]}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  locale="en"
  showAddResultsButton={false}
  toggleDraftLabOrderUrgency={[MockFunction]}
>
  <div>
    <h5
      className="h5-draft-header"
    >
      <FormattedMessage
        defaultMessage="Unsaved Draft Orders"
        description="Unsaved Draft Orders"
        id="app.orders.unsaved.draft"
        values={Object {}}
      >
        <span>
          Unsaved Draft Orders
        </span>
      </FormattedMessage>
       (
      1
      )
    </h5>
    <div
      className="table-container"
    >
      <ul
        className="draft-list-container"
      >
        <span>
          <li
            className="draft-list small-font"
            key="draft-order-undefined"
          >
            <span
              className="order-status"
            >
              NEW
            </span>
            <span
              className="draft-name"
            >
              Hct
            </span>
            <div
              className="action-btn-wrapper"
            >
              <span
                className="action-btn"
              >
                <div>
                  <IconButton
                    dataContext={
                      Object {
                        "display": "hermatocrite",
                        "names": Array [
                          Object {
                            "conceptNameType": "SHORT",
                            "id": 13207,
                            "locale": "en",
                            "localePreferred": false,
                            "name": "Hct",
                            "uuid": "0b7ba816-15f5-102d-96e4-000c29c2a5d7",
                            "voided": false,
                          },
                          Object {
                            "conceptNameType": "FULLY_SPECIFIED",
                            "id": 13204,
                            "locale": "en",
                            "localePreferred": true,
                            "name": "Hematocrit",
                            "uuid": "3e148d20-26fe-102b-80cb-0017a47871b2",
                            "voided": false,
                          },
                        ],
                        "set": true,
                        "type": "laborder",
                        "uuid": 6,
                      }
                    }
                    icon="▲"
                    iconClass="scale i-gray"
                    iconTitle="Urgency"
                    id="draft-toggle-btn icon-btn-anchor"
                    onClick={[MockFunction]}
                  >
                    <span>
                      <a
                        href="#"
                        id="draft-toggle-btn icon-btn-anchor"
                        onClick={[Function]}
                      >
                        <i
                          className="scale i-gray scale"
                          title="Urgency"
                        >
                          ▲
                        </i>
                      </a>
                    </span>
                  </IconButton>
                </div>
              </span>
              <span
                className="action-btn right"
              >
                <IconButton
                  dataContext={
                    Object {
                      "draftType": "laborder",
                      "order": Object {
                        "display": "hermatocrite",
                        "names": Array [
                          Object {
                            "conceptNameType": "SHORT",
                            "id": 13207,
                            "locale": "en",
                            "localePreferred": false,
                            "name": "Hct",
                            "uuid": "0b7ba816-15f5-102d-96e4-000c29c2a5d7",
                            "voided": false,
                          },
                          Object {
                            "conceptNameType": "FULLY_SPECIFIED",
                            "id": 13204,
                            "locale": "en",
                            "localePreferred": true,
                            "name": "Hematocrit",
                            "uuid": "3e148d20-26fe-102b-80cb-0017a47871b2",
                            "voided": false,
                          },
                        ],
                        "set": true,
                        "type": "laborder",
                        "uuid": 6,
                      },
                    }
                  }
                  icon={null}
                  iconClass="icon-remove"
                  iconTitle="Discard"
                  id="draft-discard-btn"
                  onClick={[MockFunction]}
                >
                  <span>
                    <a
                      href="#"
                      id="draft-discard-btn"
                      onClick={[Function]}
                    >
                      <i
                        className="icon-remove scale"
                        title="Discard"
                      />
                    </a>
                  </span>
                </IconButton>
              </span>
            </div>
          </li>
        </span>
      </ul>
    </div>
    <br />
    <input
      className="button  cancel modified-btn"
      id="draft-discard-all"
      onClick={[Function]}
      type="button"
      value="Discard"
    />
    <input
      className="button confirm right modified-btn"
      onClick={[Function]}
      type="submit"
      value="Save"
    />
    <br />
    <br />
  </div>
</Draft>
`;
